packages/components/src/Menu/MenuItem.tsx

```
@@ -119,7 +119,7 @@ const MenuItemInternal: FC<MenuItemProps> = (props) => {
       )
     )
 
-  const Component = itemRole === 'link' ? 'a' : 'button'
+  const Component = !disabled && itemRole === 'link' ? 'a' : 'button'
```

Ideally we should throw a `console.warn()` here explaining that `disabled` and `itemRole="link"` can't be used together so this doesn't act as a silent failure.